### ISSUE TITLE
[JIT] Canonicalize aten::rsub 

### ISSUE THREAD

**Issue by eellison** [pytorch/pytorch#64775]
## 🚀 Feature

Pytorch has [two](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/native_functions.yaml#L4844) [operators](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/native_functions.yaml#L4868) for rsub  which have the semantics `rsub(x, y) == y - x`. Generally, having two different operators that do the same thing in the your IR is not desirable because it means passes have to reason about both cases if they want to optimize the pattern. Additionally, optimizations like Common Subexpression Elimination will no trigger.

```
@torch.jit.script
def foo(x, y):
    return (x - y) + 1, (x - y) + 1

torch._C._jit_pass_cse(foo.graph)
torch._C._jit_pass_dce(foo.graph)
print(foo.graph)
```
Will give a graph with a single `aten::sub` node, where as the semantically equivalent 

```
@torch.jit.script
def foo(x: bool):
    return (x - y) + 1, torch.rsub(y, x) + 1

torch._C._jit_pass_cse(foo.graph)
torch._C._jit_pass_dce(foo.graph)
print(foo.graph)
```

Will have two different sub nodes. This is of particular interest because we have optimizations like [fusion](https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/tensorexpr_fuser.cpp#L70) which target `sub` but do not handle `rsub`. We already have an existing pass that runs whenever we emit IR to do similar normalization, we could add this pattern there. https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/normalize_ops.cpp

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
